From fa0063ba67071384d8c749cee8f4f4e5bbc8ef91 Mon Sep 17 00:00:00 2001
From: Greg Kurz <gkurz@redhat.com>
Date: Fri, 20 Nov 2020 14:00:31 -0500
Subject: redhat: Define hw_compat_8_3

RH-Author: Greg Kurz <gkurz@redhat.com>
Message-id: <20201120140033.578472-2-gkurz@redhat.com>
Patchwork-id: 99790
O-Subject: [RHEL-AV-8.4.0 qemu-kvm PATCH v2 1/3] redhat: Define hw_compat_8_3
Bugzilla: 1893935
RH-Acked-by: Danilo de Paula <ddepaula@redhat.com>
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

Signed-off-by: Greg Kurz <gkurz@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/core/machine.c   | 21 +++++++++++++++++++++
 include/hw/boards.h |  3 +++
 2 files changed, 24 insertions(+)

diff --git a/hw/core/machine.c b/hw/core/machine.c
index 19d50dde45..aba05ad676 100644
--- a/hw/core/machine.c
+++ b/hw/core/machine.c
@@ -28,6 +28,27 @@
 #include "hw/mem/nvdimm.h"
 #include "migration/vmstate.h"
 
+/*
+ * The same as hw_compat_5_1
+ */
+GlobalProperty hw_compat_rhel_8_3[] = {
+    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
+    { "vhost-scsi", "num_queues", "1"},
+    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
+    { "vhost-user-blk", "num-queues", "1"},
+    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
+    { "vhost-user-scsi", "num_queues", "1"},
+    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
+    { "virtio-blk-device", "num-queues", "1"},
+    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
+    { "virtio-scsi-device", "num_queues", "1"},
+    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
+    { "nvme", "use-intel-id", "on"},
+    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
+    { "pvpanic", "events", "1"}, /* PVPANIC_PANICKED */
+};
+const size_t hw_compat_rhel_8_3_len = G_N_ELEMENTS(hw_compat_rhel_8_3);
+
 /*
  * The same as hw_compat_4_2 + hw_compat_5_0
  */
diff --git a/include/hw/boards.h b/include/hw/boards.h
index 4e4a54b313..526e5aea04 100644
--- a/include/hw/boards.h
+++ b/include/hw/boards.h
@@ -369,6 +369,9 @@ extern const size_t hw_compat_2_2_len;
 extern GlobalProperty hw_compat_2_1[];
 extern const size_t hw_compat_2_1_len;
 
+extern GlobalProperty hw_compat_rhel_8_3[];
+extern const size_t hw_compat_rhel_8_3_len;
+
 extern GlobalProperty hw_compat_rhel_8_2[];
 extern const size_t hw_compat_rhel_8_2_len;
 
-- 
2.18.4

